### ISSUE TITLE
question about emb_dim meaning

### ISSUE THREAD

**Issue by iiiwan** [nikita-petrashen/openmax#1]
Thanks for your sharing. But what is the meaning of emb_dim = 100? Does not the out of the model is the size of class num ?

**Comment by iiiwan** [comment]
Thanks your code, I can run it. You are my academic hero and god. Thanks again. But why it is fail to use other distance like eucos and cos

**Comment by max77802** [comment]
> Thanks for your sharing. But what is the meaning of emb_dim = 100? Does not the out of the model is the size of class num ?

hi, I have the same confusion now.  So could you tell me the meaning of emb-dim? Thank you very much!!!

**Comment by nikita-petrashen** [comment]
@iiiwan thank you for the kind words
@max77802 

No, `emb_dim` is different from `n_logits`.

This approach operates in the embedding (of one of the final layers) space. In this space, embeddings are clustered by labels. Embeddings of the samples of a single class are clustered together, embeddings of different classes are far from each other. 

For a given sample, we take its embedding and see, how far is it from each label's cluster. Based on distances from the clusters, we update the sample's logits. In this approach there's no direct connection between embeddings and logits, they can even be  from a different neural network. The only important aspect is that embeddings must be clustered by class label.

Hope this helps.
